sdk/core/azure-core-test/src/main/java/com/azure/core/test/utils/TestProxyUtils.java

```diff
@@ -62,13 +58,17 @@ public class TestProxyUtils {
         "privateKey", "fencingClientPassword", "acrToken", "scriptUrlSasToken", "azureBlobSource.containerUrl",
         "properties.DOCKER_REGISTRY_SEVER_PASSWORD");
 
+    public static final String HOST_NAME_REGEX = "(?<=http://|https://)(?<host>[^/?\\\\.]+)";
     private static final List<TestProxySanitizer> HEADER_KEY_REGEX_TO_REDACT = Arrays.asList(
-
         new TestProxySanitizer("ServiceBusDlqSupplementaryAuthorization",
             "(?:(sv|sig|se|srt|ss|sp)=)(?<secret>[^&\\\"]+)", REDACTED_VALUE, TestProxySanitizerType.HEADER)
                 .setGroupForReplace("secret"),
         new TestProxySanitizer("ServiceBusSupplementaryAuthorization", "(?:(sv|sig|se|srt|ss|sp)=)(?<secret>[^&\\\"]+)",
-            REDACTED_VALUE, TestProxySanitizerType.HEADER).setGroupForReplace("secret"));
+            REDACTED_VALUE, TestProxySanitizerType.HEADER).setGroupForReplace("secret"),
+        new TestProxySanitizer("operation-location", HOST_NAME_REGEX, REDACTED_VALUE, TestProxySanitizerType.HEADER)
+            .setGroupForReplace("host"),
```

Comment:
Add a comment here explaining that we're using a different regex to keep parts of the value that the test-proxy one would entirely elide.